### PR TITLE
Team/sultans of swing/feature/ball glowing

### PR DESCRIPTION
**Pull request by kostaDimitrijevic**
Do you maybe know a reason why collision wouldn't be detected for ball? I am practicing the engine, and I made simple system... when ball collides it changes color.

### CODE REVIEW

File: source/dagger/gameplay/ping_pong/ball_glowing.cpp [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+		static bool change = false;
+		if (collision.colided)
+		{
+			glow.isColored = true;
+			collision.colided = false;
+		}
+
+		if (glow.isColored)
+		{
+			if (change) change = false;
```
Comment (DPrivalov) [inline source/dagger/gameplay/ping_pong/ball_glowing.cpp:24]: try to use {} all the time, no matter how many lines you have.

File: source/dagger/gameplay/ping_pong/ball_glowing.h [hunk 2]
```diff
@@ -0,0 +1,20 @@
+#pragma once
+#include "core/system.h"
+
+using namespace dagger;
+
+struct BallGlowing
+{
+    ColorRGBA color_for_ball;
```
Comment (DPrivalov) [inline source/dagger/gameplay/ping_pong/ball_glowing.h:8]: colorForBall

File: source/dagger/gameplay/ping_pong/ping_pong_main.cpp [hunk 3]
```diff
@@ -54,6 +56,7 @@ void PingPongGame::CoreSystemsSetup(Engine& engine_)
     engine_.AddSystem<SpriteRenderSystem>();
     engine_.AddSystem<AnimationSystem>();
     engine_.AddSystem<TransformSystem>();
+    engine_.AddSystem<BallGlowingSystem>();
```
Comment (DPrivalov) [inline source/dagger/gameplay/ping_pong/ping_pong_main.cpp:59]: move to GameplaySystemsSetup

File: source/dagger/gameplay/ping_pong/ball_glowing.cpp [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+void BallGlowingSystem::Run()
+{
+	auto view = Engine::Registry().view<BallGlowing, Sprite, SimpleCollision>();
+	for (auto entity : view) {
+		
+		auto& glow = view.get<BallGlowing>(entity);
+		auto& sprite = view.get<Sprite>(entity);
+		auto& collision = view.get<SimpleCollision>(entity);
+		static bool change = false;
```
Comment (DPrivalov) [inline source/dagger/gameplay/ping_pong/ball_glowing.cpp:15]: why is it static?

**Review by DPrivalov — CHANGES_REQUESTED** [review]

File: source/dagger/gameplay/ping_pong/ball_glowing.cpp (hunk 4 above)
Comment (kostaDimitrijevic) [reply source/dagger/gameplay/ping_pong/ball_glowing.cpp:15]: Well, I guess I wanted to keep changed value so when I enter again in run to continue with past one

File: source/dagger/gameplay/ping_pong/ball_glowing.cpp (hunk 1 above)
Comment (kostaDimitrijevic) [reply source/dagger/gameplay/ping_pong/ball_glowing.cpp:24]: Okay, I will

File: source/dagger/gameplay/ping_pong/ping_pong_main.cpp (hunk 3 above)
Comment (kostaDimitrijevic) [reply source/dagger/gameplay/ping_pong/ping_pong_main.cpp:59]: Yea... I thought that I did put it there. It's not working anyway...

**Pull request closed without merging**